src/EditorFeatures/CSharp/MisplacedUsings/MisplacedUsingsDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+            new CodeStyleOption<UsingPlacementPreference>(UsingPlacementPreference.NoPreference, NotificationOption.None);
+
+        private static readonly LocalizableResourceString s_localizableTitle = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Misplaced_using), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        private static readonly LocalizableResourceString s_localizableInsideMessage = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Using_directives_must_be_placed_inside_a_namespace_declaration), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        private static readonly LocalizableResourceString s_localizableOutsideMessage = new LocalizableResourceString(
+            nameof(CSharpEditorResources.Using_directives_must_be_placed_outside_a_namespace_declaration), CSharpEditorResources.ResourceManager, typeof(CSharpEditorResources));
+
+        internal static readonly DiagnosticDescriptor _insideDescriptor = CreateDescriptorWithId(
+            IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId,
+            s_localizableTitle,
+            s_localizableInsideMessage,
+            configurable: true);
+
+        internal static readonly DiagnosticDescriptor _outsideDescriptor = CreateDescriptorWithId(
+            IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId,
+            s_localizableTitle,
+            s_localizableOutsideMessage,
+            configurable: true);
+
+        private readonly ICodingConventionsManager _conventionsManager = CodingConventionsManagerFactory.CreateCodingConventionsManager();
+
+        public MisplacedUsingsDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.MoveMisplacedUsingsDiagnosticId, s_localizableTitle)
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.CompilationUnit);
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.NamespaceDeclaration);
+        }
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var statement = context.Node;
+            var cancellationToken = context.CancellationToken;
+
+            var option = GetPreferredPlacementOption(context);
+            if (option.Value == UsingPlacementPreference.NoPreference)
+            {
+                return;
+            }
+
+            if (statement is CompilationUnitSyntax compilationUnit
+                && option.Value == UsingPlacementPreference.InsideNamespace
+                && !ShouldSurpressDiagnostic(compilationUnit))
+            {
+                ReportDiagnostics(context, _insideDescriptor, compilationUnit.Usings, option);
+            }
+            else if (statement is NamespaceDeclarationSyntax namespaceDeclaration
+                && option.Value == UsingPlacementPreference.OutsideNamespace)
```

Comment:
can you alight parts of a conditoin with each other.  also, the vast majority of IDE code puts boolean operators on previous line.  can you be consistent with that?